storage/innobase/log/log0recv.cc

```diff
@@ -3986,6 +3986,11 @@ recv_recovery_from_checkpoint_start(
 
 	const page_id_t	page_id(max_cp_group->space_id, 0);
 
+	byte* log_hdr_buf_unalign = (byte*)ut_malloc_nokey(LOG_FILE_HDR_SIZE +
+				     OS_FILE_LOG_BLOCK_SIZE);
```

Comment:
OS_FILE_LOG_BLOCK_SIZE == 512, srv_log_write_ahead_size can be up to 65536. It looks the malloc'ed buffer will be too small in such case and after ut_align it will be overflown: srv_log_write_ahead_size must be present in malloc call too